webpack/assets/javascripts/react_app/common/urlHelpers.js

```diff
@@ -28,7 +28,8 @@ export const getURIpage = () => Number(getURI().query(true).page) || 1;
 /**
  * Get updated perPage param
  */
-export const getURIperPage = () => Number(getURI().query(true).per_page) || 20;
+export const getURIperPage = () =>
+  Number(getURI().query(true).per_page) || null;
```

Comment:
Can we use `context` here instead of `null`?

Reply:
it's a function, only components have access to the provider

Reply:
can't we just use the hooks ?

Reply:
IMO the consumer of the function should do
```js
getURIperPage() || defaultPerPage
```